pwa-devdocs/_drafts/configuration-management/index.md

```diff
@@ -0,0 +1,431 @@
+
+Like the rest of PWA Studio, Buildpack uses environment variables as its central source of configuration settings.
+A PWA using Buildpack has a [`.env` file][dotenv] in its root directory.
+This file contains configuration values as text, line by line, in the form `NAME=value`.
+In any script in any programming language, you can access these environment variables directly by sourcing the file as a legal POSIX shell script.
+
+Buildpack has a CLI for creating `.env` files and validating environments, and library methods for connecting environment management with other tools.
+Using these tools, you can keep global configuration values in a central location and propagate them throughout your project, pushing common settings down to many library functions without tightly coupling them together.
+
+## Command Line Interface
+
+The `buildpack` command is a toolkit with subcommands.
+Running it with no arguments produces this output:
+
+```text
+buildpack <command>
+
+Commands:
+  buildpack create-custom-origin            Get or create a secure, unique
+  <directory>                               hostname/port combination and a
+                                            trusted SSL certificate for local
+                                            development, which enables all PWA
+                                            features.
+  buildpack create-env-file <directory>     Generate a .env file in the provided
+                                            directory to store project
+                                            configuration as environment
+                                            variables
+  buildpack load-env <directory>            Load and validate the current
+                                            environment, including .env file if
+                                            present, to ensure all required
+                                            configuration is in place.
+
+Options:
+  --version  Show version number                                       [boolean]
+  --help     Show help                                                 [boolean]
+```
+
+A project with `@magento/pwa-buildpack` installed may use the `buildpack` command in NPM scripts.
+
+```json
+"scripts": {
+  "load-env": "buildpack load-env ."
+}
+```
+
+With such a configuration, you could run `npm run-script load-env` or `yarn run load-env` to use the local copy of the Buildpack CLI.
+
+You can also invoke `buildpack` directly using the `npx` tool, which installs packages and runs their CLIs in a single command:
+
+```sh
+npx @magento/pwa-buildpack <command>
+```
+
+_Note: It is not recommended to globally install buildpack with `yarn global add`._
+_Individual projects should use their own versions and a globally installed buildpack can cause unexpected behavior._
+
+### `buildpack create-env-file <dir>`
```

Comment:
Would be worth noting that this command expects the `MAGENTO_BACKEND_URL` environment variable to be set prior or it will create a `.env` file with a blank backend url entry.
```sh
MAGENTO_BACKEND_URL="https://magento.test/" npx buildpack create-env-file .
```

Reply:
True, we could add that there. It's not a cross-platform command (it won't work on Windows) but we could have a Windows version with `SET`.